src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter df = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromNow = LocalDate.parse(dateFrom, df);
+        LocalDate dateToNow = LocalDate.parse(dateTo, df);
+        int[] countSalary = new int [names.length];
+        for (String dataEntry : data) {
+            String[] currentStringInfo = dataEntry.split(" ");
+            int hourlyRate = Integer.parseInt(currentStringInfo[3]);
+            int hoursWorked = Integer.parseInt(currentStringInfo[2]);
+            for (int j = 0; j < names.length; j++) {
+                LocalDate today = LocalDate.parse(currentStringInfo[0], df);
+                if (names[j].equals(currentStringInfo[1]) && !dateFromNow.isAfter(today)
+                        && !(dateToNow.isBefore(today))) {
+                    countSalary[j] += hourlyRate * hoursWorked;
+                }
```

Comment:
0  1 2 3 - make the constants like HOURLY_RATE_INDEX

Reply:
i cant do this becouse this filds are fields of method NOT class